source/draw/gpu/opengl/OpenGLMap.ooc

```diff
@@ -167,125 +127,31 @@ OpenGLMapTransform: abstract class extends OpenGLMap {
 }
 OpenGLMapTransformTexture: class extends OpenGLMapTransform {
 	init: func (context: OpenGLContext) { super(This fragmentSource, context) }
-	fragmentSource: static String = "#version 300 es
-		precision highp float;
-		uniform sampler2D texture0;
-		in vec2 fragmentTextureCoordinate;
-		out vec4 outColor;
-		void main() {
-			outColor = texture(texture0, fragmentTextureCoordinate).rgba;
-		}
-		"
+	fragmentSource: static String = slurp("shaders/texture.frag")
 }
 OpenGLMapMonochromeToBgra: class extends OpenGLMapDefaultTexture {
 	init: func (context: OpenGLContext) { super(This customFragmentSource, context) }
-	customFragmentSource: static String = "#version 300 es
-		precision highp float;
-		uniform sampler2D texture0;
-		in vec2 fragmentTextureCoordinate;
-		out vec4 outColor;
-		void main() {
-			float colorSample = texture(texture0, fragmentTextureCoordinate).r;
-			outColor = vec4(colorSample, colorSample, colorSample, 1.0f);
-		}
-		"
+	customFragmentSource: static String = slurp("shaders/monochromeToBgra.frag")
 }
 OpenGLMapYuvPlanarToBgra: class extends OpenGLMapTransform {
 	init: func (context: OpenGLContext) { super(This fragmentSource, context) }
-	fragmentSource: static String = "#version 300 es
-		precision highp float;
-		uniform sampler2D texture0;
-		uniform sampler2D texture1;
-		uniform sampler2D texture2;
-		in vec2 fragmentTextureCoordinate;
-		out vec4 outColor;
-		vec4 YuvToRgba(vec4 t)
-		{
-			mat4 matrix = mat4(1, 1, 1, 0,
-				-0.000001218894189, -0.344135678165337, 1.772000066073816, 0,
-				1.401999588657340, -0.714136155581812, 0.000000406298063, 0,
-				0, 0, 0, 1);
-				return matrix * t;
-		}
-		void main() {
-			float y = texture(texture0, fragmentTextureCoordinate).r;
-			float u = texture(texture1, fragmentTextureCoordinate).r;
-			float v = texture(texture2, fragmentTextureCoordinate).r;
-			outColor = YuvToRgba(vec4(y, v - 0.5f, u - 0.5f, 1.0f));
-		}
-		"
+	fragmentSource: static String = slurp("shaders/yuvPlanarToBgra.frag")
 }
 OpenGLMapYuvSemiplanarToBgra: class extends OpenGLMapTransform {
 	init: func (context: OpenGLContext) { super(This fragmentSource, context) }
-	fragmentSource: static String = "#version 300 es
-		precision highp float;
-		uniform sampler2D texture0;
-		uniform sampler2D texture1;
-		in vec2 fragmentTextureCoordinate;
-		out vec4 outColor;
-		vec4 YuvToRgba(vec4 t)
-		{
-			mat4 matrix = mat4(1, 1, 1, 0,
-				-0.000001218894189, -0.344135678165337, 1.772000066073816, 0,
-				1.401999588657340, -0.714136155581812, 0.000000406298063, 0,
-				0, 0, 0, 1);
-				return matrix * t;
-		}
-		void main() {
-			float y = texture(texture0, fragmentTextureCoordinate).r;
-			vec2 uv = texture(texture1, fragmentTextureCoordinate).rg;
-			outColor = YuvToRgba(vec4(y, uv.r - 0.5f, uv.g - 0.5f, 1.0f));
-		}
-		"
+	fragmentSource: static String = slurp("shaders/yuvSemiplanarToBgra.frag")
 }
 OpenGLMapLines: class extends OpenGLMapTransform {
```

Comment:
Because it's used when drawing lines (but might as well be used for anything else). Don't bother with it, it will be removed in the following PR anyway (removing redundant OpenGLMap classes).